src/control/cmd/dmg/system.go

```diff
@@ -115,12 +117,18 @@ func (cmd *systemQueryCmd) Execute(_ []string) (errOut error) {
 		errOut = errors.Wrap(errOut, "system query failed")
 	}()
 
+	if cmd.AvailableOnly && (cmd.ExcludedOnly || cmd.NotJoinedOnly) || (cmd.ExcludedOnly && cmd.NotJoinedOnly) {
+		return errors.New("--show-available and --show-excluded options cannot be set together")
+	}
 	if err := cmd.validateHostsRanks(); err != nil {
 		return err
 	}
 	req := new(control.SystemQueryReq)
 	req.Hosts.Replace(&cmd.Hosts.HostSet)
 	req.Ranks.Replace(&cmd.Ranks.RankSet)
+	req.AvailableOnly = cmd.AvailableOnly
+	req.ExcludedOnly = cmd.ExcludedOnly
+	req.NotJoinedOnly = cmd.NotJoinedOnly
```

Comment:
Where is NotJoinedOnly coming from? I don't see it in the systemQueryCmd definition.

Reply:
as above